src/SimulationMap.cu

```diff
@@ -33,14 +33,23 @@ __device__ SimulationMap::SimulationMap(Polyhedron *polyhedron) :
     // Direction vector from first node to its top neighbor sets randomly
     SpacePoint direction_vector = relative_point_rotation(start_node_coordinates, start_face->get_vertices()[0],
                                                           start_face->get_normal(), M_PI * 2 * rand0to1());
-    /*
+    /**
      * Array of direction vectors from nodes with the same index
      * as in `SimulationMap::nodes` array to their top neighbors
      */
     auto *nodes_directions = (SpacePoint *)malloc(sizeof(SpacePoint));
     nodes_directions[0] = direction_vector * mapnode_dist / get_distance(direction_vector, origin);
 
-    /*
+    /**
+     * Boolean array where i-th element tells whether the i-th face have nodes or not
+     * `do_faces_have_nodes[i]` value corresponds to `polyhedron->faces[i]` face
+     */
+    auto *do_faces_have_nodes = (bool *)malloc(sizeof(bool) * polyhedron->n_of_faces);
```

Comment:
I would better name it `does_face_have_nodes`, because (if I am not mistaken) our array naming style tells to describe one element of an array, not the whole array (probably worth double-checking)